File: sdkwork-rtc-sdk-java/src/main/java/com/sdkwork/rtc/metadata/RtcDriverManager.java
package com.sdkwork.rtc.metadata;

import java.util.List;

public final class RtcDriverManager {

  public RtcProviderSelection resolveSelection(
      RtcProviderSelection.RtcProviderSelectionRequest request
  ) {
    return resolveSelection(request, RtcProviderCatalog.DEFAULT_RTC_PROVIDER_KEY);
  }

  public RtcProviderSelection resolveSelection(
      RtcProviderSelection.RtcProviderSelectionRequest request,
      String defaultProviderKey
  ) {
    return RtcProviderSelection.resolveRtcProviderSelection(request, defaultProviderKey);
  }

  public RtcProviderSupport describeProviderSupport(String providerKey) {
    var official = RtcProviderCatalog.getRtcProviderByProviderKey(providerKey).isPresent();
    var activation = RtcProviderActivationCatalog.getRtcProviderActivationByProviderKey(providerKey);

    return RtcProviderSupport.createRtcProviderSupportState(
        new RtcProviderSupport.RtcProviderSupportStateRequest(
            providerKey,
            activation
                .map(RtcProviderActivationCatalog.RtcProviderActivationCatalogEntry::builtin)
                .orElse(false),
            official,
            activation
                .map(RtcProviderActivationCatalog.RtcProviderActivationCatalogEntry::runtimeBridge)
                .orElse(false)
        )
    );
  }

  public List<RtcProviderSupport> listProviderSupport() {
    return RtcProviderCatalog.ENTRIES.stream()
        .map(entry -> describeProviderSupport(entry.providerKey()))
        .toList();
  }
}

File: sdkwork-rtc-sdk-kotlin/src/main/kotlin/com/sdkwork/rtc/metadata/RtcDriverManager.kt
package com.sdkwork.rtc.metadata

class RtcDriverManager {
    fun resolveSelection(
        request: RtcProviderSelectionRequest = RtcProviderSelectionRequest(),
        defaultProviderKey: String = RtcProviderCatalog.DEFAULT_RTC_PROVIDER_KEY,
    ): RtcProviderSelection {
        return resolveRtcProviderSelection(request, defaultProviderKey)
    }

    fun describeProviderSupport(providerKey: String): RtcProviderSupport {
        val official = getRtcProviderByProviderKey(providerKey) != null
        val activation = getRtcProviderActivationByProviderKey(providerKey)

        return createRtcProviderSupportState(
            RtcProviderSupportStateRequest(
                providerKey = providerKey,
                builtin = activation?.builtin ?: false,
                official = official,
                registered = activation?.runtimeBridge ?: false,
            )
        )
    }

    fun listProviderSupport(): List<RtcProviderSupport> {
        return RtcProviderCatalog.entries.map { describeProviderSupport(it.providerKey) }
    }
}
